REST/internal/auth/service.go

```diff
@@ -0,0 +1,87 @@
+package auth
+
+import (
+	"REST/internal/models"
+	"errors"
+
+	"golang.org/x/crypto/bcrypt"
+	"gorm.io/gorm"
+)
+
+type AuthService struct {
+	db *gorm.DB
+}
+
+// New creates a new authentication service instance.
+func New(db *gorm.DB) *AuthService {
+	return &AuthService{db: db}
+}
+
+// Register a new user.
+func (s *AuthService) Register(email, password, userType string) (*models.Base, error) {
```

Comment:
![image](https://github.com/chanakorn-aramsak/Cu-get-Ref-backend/assets/88868804/8cd8b179-89b0-44a1-860f-a4e3ae37b264)

I didn't know much but when I registered, I saw an empty field in db (Is there any way to update it later could we initiate it in the beginning (student_id) if it is no use then remove it)